dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/PublicStateManager.java

```diff
@@ -59,6 +71,12 @@ public void addListener(StateListenerInterface listener) {
         listener.setConfig(config);
     }
 
+    public void insertListener(StateListenerInterface listener, StateListenerInterface subsequent) {
```

Comment:
Could use Javadoc, alternatively I think this is just a complicated way of saying "the new listener should be before elasticlistener" so could switch `List` to `LinkedList` and expose `addFirst` and `addLast`

Reply:
Would naming it addBeforeListener() be more understandable?  In general, I'd lean against exposing internal data structures, in this case, it would make it easy to add a listener without setting its config, for example.

Reply:
Fair, or you could use the Deque interface https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Deque.html